--- a/SConscript	2014-12-17 16:03:50.000000000 +0530
+++ b/SConscript	2017-05-05 10:47:16.248832088 +0530
@@ -107,21 +107,14 @@
     env['aj_targ_headers'] = [Glob('target/' + env['TARG'] + '/*.h')]
     env['aj_targ_srcs'] = [Glob('target/' + env['TARG'] + '/*.c')]
 elif env['TARG'] in [ 'linux' ]:
-    if os.environ.has_key('CROSS_PREFIX'):
-        env.Replace(CC = os.environ['CROSS_PREFIX'] + 'gcc')
-        env.Replace(CXX = os.environ['CROSS_PREFIX'] + 'g++')
-        env.Replace(LINK = os.environ['CROSS_PREFIX'] + 'gcc')
-        env.Replace(AR = os.environ['CROSS_PREFIX'] + 'ar')
-        env['ENV']['STAGING_DIR'] = os.environ.get('STAGING_DIR', '')
-
-    if os.environ.has_key('CROSS_PATH'):
-        env['ENV']['PATH'] = ':'.join([ os.environ['CROSS_PATH'], env['ENV']['PATH'] ] )
-
-    if os.environ.has_key('CROSS_CFLAGS'):
-        env.Append(CFLAGS=os.environ['CROSS_CFLAGS'].split())
-
-    if os.environ.has_key('CROSS_LINKFLAGS'):
-        env.Append(LINKFLAGS=os.environ['CROSS_LINKFLAGS'].split())
+    env['CC'] = ARGUMENTS['CC']
+    env['CXX'] = ARGUMENTS['CXX']
+    env['LINK'] = ARGUMENTS['CC']
+    env['AR'] = ARGUMENTS['AR']
+    env['STAGING_DIR'] = ARGUMENTS['STAGING_INCDIR']
+    env['ENV']['PATH'] = os.environ['PATH']
+    env['CFLAGS'] = ARGUMENTS['CFLAGS'].split()
+    env['LINKFLAGS'] = ARGUMENTS['LINKFLAGS'].split()
 
     env['libs'] = ['rt', 'crypto', 'pthread']
     env.Append(CFLAGS=['-Wall',
