--- a/Makefile.am	2016-09-20 19:32:34.844485434 +0530
+++ b/Makefile.am	2016-09-20 19:39:34.752467776 +0530
@@ -12,7 +12,7 @@
     -I$(top_srcdir)/libFDK/include \
     -I$(top_srcdir)/libPCMutils/include
 
-AM_CXXFLAGS = -fno-exceptions -fno-rtti
+AM_CXXFLAGS = -fno-exceptions -fno-rtti -std=c++98
 libfdk_aac_la_LINK = $(LINK) $(libfdk_aac_la_LDFLAGS)
 # Mention a dummy pure C file to trigger generation of the $(LINK) variable
 nodist_EXTRA_libfdk_aac_la_SOURCES = dummy.c
