diff -Naur a/Makefile.linux b/Makefile.linux
--- a/Makefile.linux	2018-03-21 22:02:12.354495400 +0530
+++ b/Makefile.linux	2018-03-21 21:52:48.993265500 +0530
@@ -191,11 +191,7 @@
 # from the e2fsprogs
 # 'cat /proc/sys/kernel/random/uuid' could be also used
 genuuid:
-ifeq ($(TARGET_OPENWRT),)
-	sed -i -e "s/^uuid=[-0-9a-f]*/uuid=`(genuuid||uuidgen||uuid) 2>/dev/null`/" miniupnpd.conf
-else
-	sed -i -e "s/^uuid=[-0-9a-f]*/uuid=`($(STAGING_DIR_HOST)/bin/genuuid||$(STAGING_DIR_HOST)/bin/uuidgen||$(STAGING_DIR_HOST)/bin/uuid) 2>/dev/null`/" miniupnpd.conf
-endif
+	sed -i -e "s/^uuid=[-0-9a-f]*/uuid=`($(STAGING_DIR_NATIVE)/usr/bin/uuidgen) 2>/dev/null`/" miniupnpd.conf

 miniupnpd:	$(BASEOBJS) $(CPPOBJS) $(LNXOBJS)
	$(CXX) $(CPPFLAGS) $(LDFLAGS) -o $@ $(BASEOBJS) $(CPPOBJS) $(LNXOBJS) $(LDLIBS)
