diff --git a/usrsctplib/netinet/sctputil.c b/usrsctplib/netinet/sctputil.c
index 7fc8f91..3cace07 100755
--- a/usrsctplib/netinet/sctputil.c
+++ b/usrsctplib/netinet/sctputil.c
@@ -4704,6 +4704,9 @@ sctp_add_to_readq(struct sctp_inpcb *inp,
 	if (inp_read_lock_held == 0)
 		SCTP_INP_READ_LOCK(inp);
 	if (inp->sctp_flags & SCTP_PCB_FLAGS_SOCKET_CANT_READ) {
+		if (stcb && stcb->asoc.control_pdapi == control) {
+			stcb->asoc.control_pdapi = NULL;
+		}
 		sctp_free_remote_addr(control->whoFrom);
 		if (control->data) {
 			sctp_m_freem(control->data);
